### PR TITLE
UpdateSeuratFromSemla

### PR DESCRIPTION
**Pull request by jemorlanes**
helo helo.

Now the function has an argument for selecting transformed or raw images. The already written tests worked fine and I wrote an extra test for this function. It requires to use other functions like `get_array_rotation()`, but eh. This test also works fine :))

### CODE REVIEW

File: R/semla_to_seurat.R [hunk 1]
```diff
@@ -5,6 +5,7 @@ NULL
 #' Update a Seurat object created with \code{semla} for compatibility with \code{Seurat}'s spatial functions
 #' 
 #' @param object An object of class \code{Seurat} with Visium data, created with \code{semla}
+#' @param image_use Strig specifying image type to use, either "raw" or "transformed"
```
Comment (ludvigla) [inline R/semla_to_seurat.R:8]: Strig -> string

File: R/semla_to_seurat.R [hunk 2]
```diff
@@ -29,63 +30,126 @@ NULL
...
+  if (image_use == "raw") {
+    if (verbose) cli_alert_info("Loading images")
+    imgs <- lapply(st_object@imgs, function(im) {
+      ar <- image_read(im)[[1]] |> as.integer()
+      ar <- ar/max(ar)
+      return(ar)
+    })
+  } else if (image_use == "transformed") {
+    if (verbose) cli_alert_info("Converting hexadecimal colors to RGB")
+    imgs <- lapply(object@tools[["Staffli"]]@rasterlists[["transformed"]], function(im) {
```
Comment (ludvigla) [inline R/semla_to_seurat.R:98]: I think you can use `image_read` directly on the raster `im`?

**Review by ludvigla — COMMENTED** [review]
Looks nice!

File: R/semla_to_seurat.R (hunk 2 above)
Comment (jemorlanes) [reply R/semla_to_seurat.R:98]: Does it accept the raster `im` as input? I thought that you could only give a filepath

Comment (ludvigla) [reply R/semla_to_seurat.R:98]: From `image_read`:

path: `a file, url, or raster object or bitmap array`  

:-)

Comment (ludvigla) [reply R/semla_to_seurat.R:98]: No need to change if it already works I guess!

**Pull request merged**